### ISSUE TITLE
[notifications]

### ISSUE THREAD

**Issue by ardera** [cph-cachet/flutter-plugins#197]
### Device / Emulator and OS
 - Device: Xiaomi Redmi Note 8T
 - OS: Android 10

### Describe the bug
- Exception thrown when accessing `Notifications().notificationStream`

### To Reproduce
```dart
void main() {
  runApp(
    MaterialApp(
      title: 'Flutter Demo',
      home: StreamBuilder<NotificationEvent>(
        stream: Notifications().notificationStream,
        builder: (context, snapshot) {
        if (snapshot.hasData) {
          return ListTile(
            title: Text(snapshot.data.packageName),
            subtitle: Text(snapshot.data.packageMessage),
          );
        } else {
          return Text("none");
        }
      })
    )
  );
}
```

### Expected behavior
Expected would be the app displaying the latest notifications' `packageName` and `packageMessage`, or alternatively the text `none` when no notification has arrived yet.

### Actual behavior
```
Launching lib\main.dart on Redmi Note 8T in debug mode...
✓ Built build\app\outputs\flutter-apk\app-debug.apk.
Installing build\app\outputs\flutter-apk\app.apk...
Connecting to VM Service at ws://127.0.0.1:50999/VfgEg4cUn7M=/ws
W/c_companion_ap(30115): Accessing hidden method Landroid/view/accessibility/AccessibilityNodeInfo;->getSourceNodeId()J (light greylist, reflection)
W/c_companion_ap(30115): Accessing hidden method Landroid/view/accessibility/AccessibilityRecord;->getSourceNodeId()J (light greylist, reflection)
W/c_companion_ap(30115): Accessing hidden field Landroid/view/accessibility/AccessibilityNodeInfo;->mChildNodeIds:Landroid/util/LongArray; (light greylist, reflection)
W/c_companion_ap(30115): Accessing hidden method Landroid/util/LongArray;->get(I)J (light greylist, reflection)
W/Activity(30115): Slow Operation: Activity com.example.sharandac_companion_app/.MainActivity onCreate took 4024ms
I/OpenGLRenderer(30115): Initialized EGL, version 1.4
D/OpenGLRenderer(30115): Swap behavior 2
E/EventChannel#notifications.eventChannel(30115): Failed to open event stream
E/EventChannel#notifications.eventChannel(30115): android.util.AndroidRuntimeException: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?
E/EventChannel#notifications.eventChannel(30115): 	at android.app.ContextImpl.startActivity(ContextImpl.java:922)
E/EventChannel#notifications.eventChannel(30115): 	at android.app.ContextImpl.startActivity(ContextImpl.java:898)
E/EventChannel#notifications.eventChannel(30115): 	at android.content.ContextWrapper.startActivity(ContextWrapper.java:389)
E/EventChannel#notifications.eventChannel(30115): 	at dk.cachet.notifications.NotificationsPlugin.onListen(NotificationsPlugin.kt:46)
E/EventChannel#notifications.eventChannel(30115): 	at io.flutter.plugin.common.EventChannel$IncomingStreamRequestHandler.onListen(EventChannel.java:188)
E/EventChannel#notifications.eventChannel(30115): 	at io.flutter.plugin.common.EventChannel$IncomingStreamRequestHandler.onMessage(EventChannel.java:167)
E/EventChannel#notifications.eventChannel(30115): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
E/EventChannel#notifications.eventChannel(30115): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:692)
E/EventChannel#notifications.eventChannel(30115): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/EventChannel#notifications.eventChannel(30115): 	at android.os.MessageQueue.next(MessageQueue.java:326)
E/EventChannel#notifications.eventChannel(30115): 	at android.os.Looper.loop(Looper.java:165)
E/EventChannel#notifications.eventChannel(30115): 	at android.app.ActivityThread.main(ActivityThread.java:6864)
E/EventChannel#notifications.eventChannel(30115): 	at java.lang.reflect.Method.invoke(Native Method)
E/EventChannel#notifications.eventChannel(30115): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
E/EventChannel#notifications.eventChannel(30115): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:873)

════════ Exception caught by services library ══════════════════════════════════
The following PlatformException was thrown while activating platform stream on channel notifications.eventChannel:
PlatformException(error, Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?, null)

When the exception was thrown, this was the stack
#0      StandardMethodCodec.decodeEnvelope package:flutter/…/services/message_codecs.dart
#1      MethodChannel._invokeMethod package:flutter/…/services/platform_channel.dart
<asynchronous suspension>
#2      MethodChannel.invokeMethod package:flutter/…/services/platform_channel.dart
#3      EventChannel.receiveBroadcastStream.<anonymous closure> package:flutter/…/services/platform_channel.dart
...
════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY SERVICES LIBRARY ╞══════════════════════════════════════════════════════════
The following PlatformException was thrown while activating platform stream on channel
notifications.eventChannel:
PlatformException(error, Calling startActivity() from outside of an Activity  context requires the
FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?, null)
When the exception was thrown, this was the stack:
#0      StandardMethodCodec.decodeEnvelope package:flutter/…/services/message_codecs.dart
#1      MethodChannel._invokeMethod package:flutter/…/services/platform_channel.dart
<asynchronous suspension>
#2      MethodChannel.invokeMethod package:flutter/…/services/platform_channel.dart
#3      EventChannel.receiveBroadcastStream.<anonymous closure> package:flutter/…/services/platform_channel.dart
#4      EventChannel.receiveBroadcastStream.<anonymous closure> package:flutter/…/services/platform_channel.dart
#12     _StreamBuilderBaseState._subscribe package:flutter/…/widgets/async.dart
#13     _StreamBuilderBaseState.initState package:flutter/…/widgets/async.dart
#14     StatefulElement._firstBuild package:flutter/…/widgets/framework.dart
#15     ComponentElement.mount package:flutter/…/widgets/framework.dart
...     Normal element mounting (132 frames)
#147    Element.inflateWidget package:flutter/…/widgets/framework.dart
#148    MultiChildRenderObjectElement.mount package:flutter/…/widgets/framework.dart
...     Normal element mounting (267 frames)
#415    Element.inflateWidget package:flutter/…/widgets/framework.dart
#416    Element.updateChild package:flutter/…/widgets/framework.dart
#417    RenderObjectToWidgetElement._rebuild package:flutter/…/widgets/binding.dart
#418    RenderObjectToWidgetElement.mount package:flutter/…/widgets/binding.dart
#419    RenderObjectToWidgetAdapter.attachToRenderTree.<anonymous closure> package:flutter/…/widgets/binding.dart
#420    BuildOwner.buildScope package:flutter/…/widgets/framework.dart
#421    RenderObjectToWidgetAdapter.attachToRenderTree package:flutter/…/widgets/binding.dart
#422    WidgetsBinding.attachRootWidget package:flutter/…/widgets/binding.dart
#423    WidgetsBinding.scheduleAttachRootWidget.<anonymous closure> package:flutter/…/widgets/binding.dart
(elided 18 frames from class _RawReceivePortImpl, class _Timer, dart:async, and dart:async-patch)
════════════════════════════════════════════════════════════════════════════════════════════════════
I/c_companion_ap(30115): ProcessProfilingInfo new_methods=602 is saved saved_to_disk=1 resolve_classes_delay=8000
```

### Flutter doctor 
```
> flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 1.20.4, on Microsoft Windows [Version 10.0.20221.1000], locale de-DE)
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    X Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit
      https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions.
[!] Android Studio (not installed)
[√] VS Code, 64-bit edition (version 1.49.1)
[√] Connected device (1 available)

! Doctor found issues in 2 categories.
```

**Comment by thomasnilsson** [comment]
I will look into this. Never had such an issue before!

**Comment by thomasnilsson** [comment]
Solved in version 1.0.2+1

**Comment by ardera** [comment]
Thanks!
